### ISSUE TITLE
Add warnings when environment variables change and wont be honored

### ISSUE THREAD

**Issue by dcbaker** [mesonbuild/meson#5223]
We only honor most environment variables in `meson setup`, and not in `meson configure` or ninja, with #4931 that extends to pkg-config as well. We really should have a warning when we detect that those environment variables have changed and wont be honored, at least for `meson configure`.

**Comment by jon-turney** [comment]
I don't think this is closed by #6528, since there are env vars other than PKG_CONFIG_PATH which effect the behaviour of meson?

**Comment by xclaesse** [comment]
Good point, reopening.
